qiime2/metadata/metadata.py

```diff
@@ -135,10 +135,12 @@ def _validate_pandas_index(self, index, label):
                     "Detected metadata %s that conflicts with a name reserved "
                     "for ID headers: %r" % (label, value))
 
-        # TODO use qiime2.core.util.find_duplicates in error message
-        if len(index) != len(set(e.lower() for e in index)):
+        if len(index) != len(set(index)):
+            duplicates = find_duplicates(index)
             raise ValueError(
-                "Detected duplicate metadata %s(s) (case-insensitive)" % label)
+                "Metadata %ss must be unique. The following %ss are "
+                "duplicated: %s" %
+                (label, label, ', '.join(repr(e) for e in sorted(duplicates))))
```

Comment:
I know it's not that trivial and this should not be a blocker but what about also adding the column numbers where those labels are found?

Reply:
That's a good idea, I'll put it on my TODO list. If it's not something I get to in the near-term (i.e. within the next week as I wrap up the Metadata project), I'll create an issue so we can track this feature.